pkg/backend/httpstate/backend.go

```diff
@@ -480,33 +480,41 @@ func (b *cloudBackend) Name() string {
 }
 
 func (b *cloudBackend) URL() string {
-	user, _, err := b.CurrentUser()
+	user, err := b.currentUser(context.Background())
 	if err != nil {
 		return cloudConsoleURL(b.url)
 	}
-	return cloudConsoleURL(b.url, user)
+	return cloudConsoleURL(b.url, user.Username)
 }
 
 func (b *cloudBackend) SetCurrentProject(project *workspace.Project) {
 	b.currentProject = project
 }
 
-func (b *cloudBackend) CurrentUser() (string, []string, error) {
-	return b.currentUser(context.Background())
+func (b *cloudBackend) CurrentUser() (*workspace.Account, error) {
+	account, err := b.currentUser(context.Background())
+	if err != nil {
+		return nil, err
+	}
+	return &account, nil
 }
 
-func (b *cloudBackend) currentUser(ctx context.Context) (string, []string, error) {
+func (b *cloudBackend) currentUser(ctx context.Context) (workspace.Account, error) {
 	account, err := workspace.GetAccount(b.CloudURL())
 	if err != nil {
-		return "", nil, err
+		return workspace.Account{}, err
 	}
+
 	if account.Username != "" {
-		logging.V(1).Infof("found username for access token")
-		return account.Username, account.Organizations, nil
+		logging.V(1).Infof("no username for access token")
+		name, orgs, err := b.client.GetPulumiAccountDetails(ctx)
+		if err != nil {
+			return workspace.Account{}, err
+		}
+		account.Username = name
```

Comment:
Good spot

Reply:
Sorry, clarify? Are we not saving the account information here?

Also, just looking at the implementation in here, I'm not sure we should use GetAccount here? That'll load the credentials from disk again which (besides being a leak of information from the workspace into a backend which we may or may not care about right now) was probably already done once before by the time we got here in the backend (because we probably had to construct a client).
I may be missing something but it looks like we can just unconditionally call GetPulumiAccountDetails—it'll record the user information it gets back from the server.

Reply:
>Sorry, clarify? Are we not saving the account information here? 

No we'd need a StoreAccount call, which we do on login or credential refresh.

But yeh I'll look into this a bit more, I suspect you might be right that we'll already have an account from when we set the backend up.